va/va.h

```diff
@@ -2003,6 +2015,142 @@ typedef struct _VAEncMiscParameterParallelRateControl {
+            uint32_t reserved                                : 1;
+        } bits;
+         uint8_t value;
+    } inter_sub_mb_partition_mask;
+
+    /**
+     * \brief Precison of motion search
+     * 0:Integer mode searching
+     * 1:Half-pel mode searching
+     * 2:Reserved
+     * 3:Quarter-pel mode searching
+     */
+    uint32_t enable_sub_pel_mode;
+    uint8_t sub_pel_mode;
+} VAEncMiscParameterSubMbPartPel;
+
+// per frame encoder controls, once set they will persist for all future frames
+// till it is updated again.
+typedef struct _VAEncMiscParameterEncQuality
+{
+    union
+    {
+        struct
+        {
+            // Use raw frames for reference instead of reconstructed frames.
+            uint32_t useRawPicForRef                    : 1;
+            // Disables skip check for ENC.
+            uint32_t skipCheckDisable                   : 1;
+            // Indicates app will override default driver FTQ settings using FTQEnable.
+            uint32_t FTQOverride                        : 1;
+            // Enables/disables FTQ.
+            uint32_t FTQEnable                          : 1;
+            // Indicates the app will provide the Skip Threshold LUT to use when FTQ is
+            // enabled (FTQSkipThresholdLUT), else default driver thresholds will be used.
+            uint32_t FTQSkipThresholdLUTInput           : 1;
+            // Indicates the app will provide the Skip Threshold LUT to use when FTQ is
+            // disabled (NonFTQSkipThresholdLUT), else default driver thresholds will be used.
+            uint32_t NonFTQSkipThresholdLUTInput        : 1;
+            uint32_t ReservedBit                        : 1;
+            // Control to enable the ENC mode decision algorithm to bias to fewer B Direct/Skip types.
+            // Applies only to B frames, all other frames will ignore this setting.
+            uint32_t directBiasAdjustmentEnable         : 1;
+            // Enables global motion bias.
+            uint32_t globalMotionBiasAdjustmentEnable   : 1;
+            // MV cost scaling ratio for HME predictors.  It is used when
+            // globalMotionBiasAdjustmentEnable == 1, else it is ignored.  Values are:
+            //      0: set MV cost to be 0 for HME predictor.
+            //      1: scale MV cost to be 1/2 of the default value for HME predictor.
+            //      2: scale MV cost to be 1/4 of the default value for HME predictor.
+            //      3: scale MV cost to be 1/8 of the default value for HME predictor.
+            uint32_t HMEMVCostScalingFactor             : 2;
+            //disable HME
+            uint32_t HMEDisable                         : 1;
+            //disable Super HME
+            uint32_t SuperHMEDisable                    : 1;
+            //disable Ultra HME
+            uint32_t UltraHMEDisable                    : 1;
+            //disable panic mode
+            uint32_t PanicModeDisable                   : 1;
```

Comment:
What is panic mode? 
Is this Intel-specific?

Reply:
panic mode is not intel-specific, for the HRD , if BRC need a very small size frame, and the QP is 51 already. it may trigger panic , discarding all AC coefficients (CBP=0),  it will reduce a lot of detail information. I will add comment for this mode, I also can  change the parameter name to discardCoffforBRC  ect.

Reply:
Got it, thanks. Please add more comments on fields like this. There are other vendors who have working drivers on top of VA-API :) How about "BrcPanicModeDisable"(if it is only for BRC). ?
Honestly, I am fine with any name if there are enough comments explaining what it means.

Reply:
ok, will update the comments, thanks